File: C++/2150. Find All Lonely Numbers in the Array.cpp
class Solution {
public:
    vector<int> findLonely(vector<int>& n) {
        vector<int>a;
        sort(n.begin(),n.end());
        for(int i=0;i<n.size();i++){
            if((i==0||n[i]-n[i-1]>1)&&(i==n.size()-1||n[i+1]-n[i]>1))
            a.push_back(n[i]);
        }
        return a;
    }
};

File: JAVA/2150. Find All Lonely Numbers in the Array.java
class Solution {
    public List<Integer> findLonely(int[] n) {
        List<Integer>a=new ArrayList<>();
        Arrays.sort(n);
        for(int i=0;i<n.length;i++){
            if((i==0||n[i]-n[i-1]>1)&&(i==n.length-1||n[i+1]-n[i]>1))
            a.add(n[i]);
        }
        return a;
    }
}